classes/plain.lua

```diff
@@ -174,22 +184,16 @@ SILE.registerCommand("hbox", function (_, content)
       depth = d,
       value = recentContribution,
       outputYourself = function (self, typesetter, line)
-        -- Yuck!
-        if typesetter.frame:writingDirection() == "RTL" then
-          typesetter.frame:advanceWritingDirection(self:scaledWidth(line))
-        end
+        local _post = _rtl_pre_post(self, typesetter, line)
```

Comment:
This assignment is (conditionally!) being used for its side effect, which is arguably more ugly than the original method! Just a note to remind to try to think of a way to make this clearer.

Reply:
I'm not sure I would call this a side effect. It's stashing the result of an operation in a callback to that it can be checked later without redoing the operation. Perhaps better variable names would work. I actually tried 3 or 4 things before I settled on this, it really wasn't clear to me what was happening at first, so I was probably cross-eyed by the time I came up with this.